src/increment_version.rs

```diff
@@ -0,0 +1,182 @@
+  /// - [`sysexits::ExitCode::DataErr`]
+  /// - [`sysexits::ExitCode::Unavailable`]
+  /// - [`sysexits::ExitCode::Usage`]
+  pub fn main(&self) -> Result<()> {
+    self.wrap().main()
+  }
+
+  fn wrap(&self) -> Logic {
+    Logic {
+      cli: self.clone(),
+      new_version: String::new(),
+      old_version: self
+        .old_version
+        .strip_prefix('v')
+        .unwrap_or(&self.old_version)
+        .to_string(),
+    }
+  }
+}
+
+struct Logic {
+  cli: IncrementVersion,
+  new_version: String,
+  old_version: String,
+}
+
+impl Logic {
+  fn determine_new_version(&mut self) -> Result<()> {
+    self.new_version = Version::from_str(&self.old_version)?
+      .increment(self.cli.range)
+      .to_string();
+    self.new_version = self
+      .new_version
+      .strip_prefix('v')
+      .unwrap_or(&self.new_version)
+      .to_string();
+    Ok(())
+  }
+
+  fn main(&mut self) -> Result<()> {
+    self.determine_new_version()?;
+
+    for file in &self.cli.file_to_edit {
+      match file
+        .file_name()
+        .ok_or(ExitCode::Usage)?
+        .to_str()
+        .ok_or(ExitCode::DataErr)?
+      {
+        "Cargo.lock" => self.update_cargo_lock(file)?,
+        "Cargo.toml" => self.update_cargo_toml(file)?,
+        _ => self.update_normal_file(file)?,
+      }
+    }
+
+    Ok(())
+  }
+
+  fn update_cargo_lock(&self, file: &PathBuf) -> Result<()> {
+    if let Some(rust_package) = &self.cli.rust_package {
+      let mut edited = false;
+      let mut lock_file = match cargo_lock::Lockfile::load(file) {
+        Ok(l) => Ok(l),
+        Err(cargo_lock::Error::Io(e)) => Err(e.into()),
+        Err(_) => Err(ExitCode::Unavailable),
+      }?;
+
+      for package in &mut lock_file.packages {
+        if package.name.as_str() == rust_package {
+          match self.cli.range {
+            VersionRange::Major => {
+              package.version.major += 1;
+              package.version.minor = 0;
+              package.version.patch = 0;
+            }
+            VersionRange::Minor => {
+              package.version.minor += 1;
+              package.version.patch = 0;
+            }
+            VersionRange::Patch => package.version.patch += 1,
+          }
+
+          edited = true;
+          break;
+        }
+      }
+
+      if edited {
+        file.truncate(Box::new(lock_file.clone().to_string()))
+      } else {
+        Err(ExitCode::DataErr)
+      }
+    } else {
+      Err(ExitCode::Usage)
+    }
+  }
+
+  fn update_cargo_toml(&self, file: &PathBuf) -> Result<()> {
+    if let Ok(mut manifest) =
+      file.read()?.try_into_string()?.parse::<toml::Table>()
+    {
+      if manifest["package"]["version"].as_str().is_some() {
+        manifest["package"]["version"] = self.new_version.clone().into();
+        file.truncate(Box::new(manifest.to_string()))?;
+
+        Ok(())
+      } else {
+        Err(ExitCode::DataErr)
+      }
+    } else {
+      Err(ExitCode::DataErr)
+    }
+  }
```

Comment:
*Any* file named `Cargo.toml` will be updated as follows:

- Aeruginous searches for the `package.version` field and updates exclusively that value.
- If no version field is present in the respective `Cargo.toml`, the update will fail.

Reply:
@kevinmatthes 
I'm afraid that overwriting the whole file would lead to remove any comments written inside the cargo.toml file or any commented-out configs. I ran into the same problem with Tui-Journal as I wanted to rewrite the config file there.

I suggest using the old logic (Changing the version manually) + Verifying the changes with toml parser would be enough for a simple use-case.

Increasing the version from the CLI seems to be a complicated issue [isse](https://github.com/rust-lang/cargo/issues/6583)

Reply:
I will add another mode:  rewriting.  In contrast to editing files to only change the necessary lines, rewriting shall also reformat the file.